pandas/io/formats/format.py

```diff
@@ -894,6 +894,8 @@ def format_array(values, formatter, float_format=None, na_rep='NaN',
         fmt_klass = ExtensionArrayFormatter
     elif is_float_dtype(values.dtype):
```

Comment:
you can just do this in an or:

``is_float_dtype(...) or is_complex_dtype(...)`` easier to read